update/stock_update.js

```diff
@@ -1,122 +1,128 @@
-//---CONFIGURE AUTHENTICATION HERE--------------------
-var CLOUDANT_KEY      = 'XXXXXXXXX';
-var CLOUDANT_PASSWORD = 'YYYYYYYYY';
-var CLOUDANT_ACCESS   = 'someurl-bluemix.cloudant.com'
-var DB_NAME = 'DB_NAME';
-//----------------------------------------------------
 
-var Cloudant = require('cloudant');
-
-var cloudant = Cloudant({
-    account: CLOUDANT_ACCESS,
-    key: CLOUDANT_KEY,
-    password: CLOUDANT_PASSWORD
-});
-var db = cloudant.db.use(DB_NAME);
+const stock_db = require('./stock_db.js');
+const utils = require('./utils.js');
+const discovery = require('./discovery.js');
 
 //TODO these should be the companies' tickers
 var companies = ['A', 'B', 'C', 'D'];
 
-function isFunc(func) {
-    return typeof func == 'function';
-}
-
-function getDocs(callback) {
-    db.list({
-        //this field is needed to return all doc data
-        include_docs: true
-    }, function(err, data) {
-        if (isFunc(callback)) {
-            var docs = data.rows.map(function(row) {
-                return row.doc;
-            });
-            callback(err, docs);
-        }
-    });
-}
-
-//include _id and _rev of existing doc in the doc to perform an update
-function insertOrUpdateDoc(doc, callback) {
-
-    db.insert(doc, function(err, body, header) {
-        if (err) {
-            console.log('insertion failed: ' + err.message);
-        } else if (isFunc(callback)) {
-            callback(body, header);
-        }
-    });
-}
-
 function findStockDatum(stocks, company) {
 
-    for (var i = 0; i < stocks.length; i++) {
-        var stock = stocks[i];
-        if (stock.ticker == company) {
-            return stock;
-        }
+  for (var i = 0; i < stocks.length; i++) {
+    var stock = stocks[i];
+    if (stock.ticker == company) {
+      return stock;
     }
+  }
 
-    return undefined;
+  return undefined;
 }
 
 function updateStocksData(articleData, stockData) {
+  
+  for (var i = 0; i < articleData.length; i++) {
+    var articleDatum = articleData[i];
+    var company = articleDatum.company
+    var articles = articleDatum.articles;
+    var stockDatum = findStockDatum(stockData, company);
+    if (stockDatum) {
+      stockDatum.history = articles.concat(stockDatum.history);
+    } else {
+      stockDatum = {
+        ticker : company,
+        history : articles
+      };
+    }
+    
+    //TODO batch insert?
+    console.log('Inserting into company: ' + company + ' articles: ' );
+    console.log(articles);
+    console.log();
+    stock_db.insertOrUpdateDoc(stockDatum);
+  }
+}
 
-    for (var i = 0; i < articleData.length; i++) {
-        var articleDatum = articleData[i];
-        var stockDatum = findStockDatum(stockData, articleDatum.company);
+function parseArticle(result) {
+  return {
+    url: result.url,
+    sentiment: result.enriched_text.sentiment.document.label,
```

Comment:
hmmm `label`? i think you want `score` too? label is just a binary value, score would be handy too.

```
        "enriched_text": {
          "sentiment": {
            "document": {
              "score": 0.260014,
              "label": "positive"
            }
          },
```

https://www.ibm.com/watson/developercloud/discovery/api/v1/#test-config

Reply:
Sure, was just a starting point.